tools/clang/lib/SPIRV/SpirvContext.cpp

```diff
@@ -221,6 +221,12 @@ const SpirvPointerType *SpirvContext::getPointerType(const SpirvType *pointee,
       return foundSC->second;
   }
 
+  if (spvTargetEnv >= SPV_ENV_VULKAN_1_2) {
```

Comment:
this is actually a bug. You are checking the type cache using the old storage class, and the inserting into the cache using the new storage class.